pipeline-model-extensions/src/main/java/org/jenkinsci/plugins/pipeline/modeldefinition/model/BuildCondition.java

```diff
@@ -77,9 +81,15 @@ protected final Result combineResults(@Nonnull WorkflowRun run) {
         return combineResults(run, null);
     }
 
+    @Deprecated
     @Nonnull
     protected final Result combineResults(@Nonnull WorkflowRun run, @CheckForNull Throwable error) {
-        return BuildCondition.getCombinedResult(run, error);
+        return combineResults(run, error, null, null);
+    }
+
+    @Nonnull
+    protected final Result combineResults(@Nonnull WorkflowRun run, @CheckForNull Throwable error, @CheckForNull FlowNode startNode, @CheckForNull FlowNode endNode) {
```

Comment:
```suggestion
    protected final Result combineResults(@Nonnull WorkflowRun run, @CheckForNull Throwable error, @CheckForNull Object context) {
```

Reply:
Wish I could, but `Stage` is in `pipeline-model-definition`, so we're kinda stuck with this.

Reply:
Well, then make a static method over in `pipeline-model-definition`?  Or a new base class local to `pipeline-model-definition`?   There are options.

Reply:
We need `BuildCondition` to be here so that it can theoretically be used to implement new conditions outside of Declarative proper, so...yeah. This is what it is. I regret the way things are organized here, but I don't see this as near bad enough to merit the refactoring involved and potential impact.

Reply:
okay, I'm not going to block on this, but if the advantage is purely theoretical and causes code smells we should keep track of it as an issue for the theoretical future where we take the time to fix design issues.

Reply:
The idea is that the extension points, such as `BuildCondition`, should only need to depend on `pipeline-model-extensions`, not `pipeline-model-definition`, and shouldn't need to know anything about the runtime or parse-time models Declarative uses (with the exception of `DeclarativeValidatorContributor` needing to know about the various `ModelAST*` classes for fairly obvious reasons) - so the more I think about it, the more I think that passing `FlowNode`s to `#combineResults` is actually preferable to passing a `Stage`. That lets the `BuildCondition` implementations not be adversely affected by changes to the runtime model - `FlowNode` is basically a primitive construct here, while `Stage` carries a lot more specific meaning that we might want to alter in the future.